aws-s3-transfer-manager/src/operation/upload.rs

```diff
@@ -14,6 +14,7 @@ mod service;
 
 use crate::error;
 use crate::io::InputStream;
+use aws_sdk_s3::primitives::ByteStream;
```

Comment:
prefer `ByteStream` from `aws-smithy-types` (the one from S3 is the same, it's a re-export of `aws_smithy_types::byte_stream::ByteStream`)

Reply:
Hmm, I tend to think the opposite; for whatever is re-exported from S3, the transfer manager should use those re-exports. Ideally, the transfer manager is a layer on top of `aws-sdk-s3` and does not need to directly depend on `aws-smithy-types` to use a skip dependency 🤔 (for the same reason AWS SDKs make use of re-exports as much as possible)

Reply:
Hmm, I tend to think the opposite; for whatever is re-exported from S3, the transfer manager should use those re-exports. Ideally, the transfer manager is a layer on top of `aws-sdk-s3` and does not need to directly depend on `aws-smithy-types` to use a skip dependency 🤔 (for the same reason AWS SDKs make use of re-exports as much as possible)

In fact, if there is a dependency in `aws-smithy-types` (or other types from Smithy crates) that the transfer manager needs to reach out for, we should re-export it from `aws-sdk-s3`.

Reply:
Spoke to Aaron offline. Taking the this stance for now,
> if we see it as being on the same level, it could directly depend on the Smithy runtime crates.

let's use the one from `aws_smithy_types` for now. I'll do the same for my PRs for consistency. Should be a two-way door; if we want to switch to re-exports for some reason in the future, we should be able to do so without breaking public APIs.

Reply:
Spoke to Aaron offline. Taking this stance for now,
> if we see it as being on the same level, it could directly depend on the Smithy runtime crates.

let's use the one from `aws_smithy_types` for now. I'll do the same for my PRs for consistency. Should be a two-way door; if we want to switch to re-exports for some reason in the future, we should be able to do so without breaking public APIs.